04-onion-routing.md

```diff
@@ -292,7 +293,7 @@ The writer:
     - otherwise:
       - MUST set `total_msat` to the amount it wishes to pay.
     - MUST ensure that the total `amount_msat` of the HTLC set which arrives at the payee
-      is equal to `total_msat`.
+      is equal to or greater than `total_msat`.
```

Comment:
I think it is worth considering the consequences of doing this without a feature bit in the invoice to signal the recipient supports such overpaying. Although possibly rare, the sender cannot know if he can overpay `total_msat` and it will be accepted.

Is it overkill to signal in the invoice whether we support such overpayment?

Reply:
That's a good point. I'm not sure it's worth "wasting" a feature bit, but maybe a simple flag in the invoice would be enough? I'll look into it.

Reply:
Perhaps the two changes in this PR can be decoupled? The relaxation of `amt_to_fwd` seems fairly straight-forward, where as changing the mpp logic probably needs further discussion.

Reply:
They are actually already somewhat decoupled since I'm just building on top of Matt's PR.
I can rebase my changes directly on top of `master` though if needed, let's wait for other people to chime in first.